packages/lexical-yjs/src/SyncCursors.ts

```diff
@@ -295,50 +295,75 @@ function updateCursor(
   }
 }
 
-export function $syncLocalCursorPosition(
-  binding: Binding,
-  provider: Provider,
-): void {
-  const awareness = provider.awareness;
-  const localState = awareness.getLocalState();
+type AnyCollabNode =
+  | CollabDecoratorNode
+  | CollabElementNode
+  | CollabTextNode
+  | CollabLineBreakNode;
 
-  if (localState === null) {
-    return;
-  }
+export function getAnchorAndFocusCollabNodesForUserState(
+  binding: Binding,
+  userState: UserState,
+) {
+  const {anchorPos, focusPos} = userState;
 
-  const anchorPos = localState.anchorPos;
-  const focusPos = localState.focusPos;
+  let anchorCollabNode: AnyCollabNode | null = null;
+  let anchorOffset = 0;
+  let focusCollabNode: AnyCollabNode | null = null;
+  let focusOffset = 0;
 
   if (anchorPos !== null && focusPos !== null) {
     const anchorAbsPos = createAbsolutePosition(anchorPos, binding);
     const focusAbsPos = createAbsolutePosition(focusPos, binding);
 
     if (anchorAbsPos !== null && focusAbsPos !== null) {
-      const [anchorCollabNode, anchorOffset] = getCollabNodeAndOffset(
+      [anchorCollabNode, anchorOffset] = getCollabNodeAndOffset(
         anchorAbsPos.type,
         anchorAbsPos.index,
       );
-      const [focusCollabNode, focusOffset] = getCollabNodeAndOffset(
+      [focusCollabNode, focusOffset] = getCollabNodeAndOffset(
         focusAbsPos.type,
         focusAbsPos.index,
       );
+    }
+  }
 
-      if (anchorCollabNode !== null && focusCollabNode !== null) {
-        const anchorKey = anchorCollabNode.getKey();
-        const focusKey = focusCollabNode.getKey();
+  return {
+    anchorCollabNode,
+    anchorOffset,
+    focusCollabNode,
+    focusOffset,
+  };
+}
 
-        const selection = $getSelection();
+export function $syncLocalCursorPosition(
+  binding: Binding,
+  provider: Provider,
+): void {
+  const awareness = provider.awareness;
+  const localState = awareness.getLocalState();
 
-        if (!$isRangeSelection(selection)) {
-          return;
-        }
-        const anchor = selection.anchor;
-        const focus = selection.focus;
+  if (localState === null) {
+    return;
+  }
 
-        $setPoint(anchor, anchorKey, anchorOffset);
-        $setPoint(focus, focusKey, focusOffset);
-      }
+  const {anchorCollabNode, anchorOffset, focusCollabNode, focusOffset} =
+    getAnchorAndFocusCollabNodesForUserState(binding, localState);
+
+  if (anchorCollabNode !== null && focusCollabNode !== null) {
+    const anchorKey = anchorCollabNode.getKey();
+    const focusKey = focusCollabNode.getKey();
+
+    const selection = $getSelection();
+
+    if (!$isRangeSelection(selection)) {
+      return;
     }
+    const anchor = selection.anchor;
+    const focus = selection.focus;
+
+    $setPoint(anchor, anchorKey, anchorOffset);
+    $setPoint(focus, focusKey, focusOffset);
```

Comment:
No reason to have variables for these points, just adds more code

```suggestion
    $setPoint(selection.anchor, anchorKey, anchorOffset);
    $setPoint(selection.focus, focusKey, focusOffset);
```